enterprise/trackedusers/data_collector.go

```diff
@@ -0,0 +1,192 @@
+package trackedusers
+
+//go:generate mockgen -destination=./mocks/mock_data_collector.go -package=mockdatacollector github.com/rudderlabs/rudder-server/enterprise/trackedusers DataCollector
+
+import (
+	"context"
+	"database/sql"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	migrator "github.com/rudderlabs/rudder-server/services/sql-migrator"
+
+	obskit "github.com/rudderlabs/rudder-observability-kit/go/labels"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+
+	"github.com/lib/pq"
+
+	"github.com/spaolacci/murmur3"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/segmentio/go-hll"
+
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	txn "github.com/rudderlabs/rudder-server/utils/tx"
+)
+
+const (
+	idTypeUserID                       = "userID"
+	idTypeAnonymousID                  = "anonymousID"
+	idTypeUserIDAnonymousIDCombination = "userIDAnonymousIDCombination"
+
+	// changing this will be non backwards compatible
+	murmurSeed = 123
+)
+
+// DataCollector is interface to collect data from jobs
+type DataCollector interface {
+	CollectData(ctx context.Context, jobs []*jobsdb.JobT, tx *txn.Tx) error
+}
+
+type UniqueUsersCollector struct {
+	log         logger.Logger
+	hllSettings *hll.Settings
+	instanceID  string
+}
+
+func NewUniqueUsersCollector(log logger.Logger, dbConn string) (*UniqueUsersCollector, error) {
+	dbHandle, err := sql.Open("postgres", dbConn)
+	if err != nil {
+		panic(err)
+	}
+	dbHandle.SetMaxOpenConns(1)
+
+	m := &migrator.Migrator{
+		Handle:                     dbHandle,
+		MigrationsTable:            "tracked_users_reports_migrations",
+		ShouldForceSetLowerVersion: config.GetBool("SQLMigrator.forceSetLowerVersion", true),
+	}
+	err = m.Migrate("tracked_users")
+	if err != nil {
+		return nil, fmt.Errorf("could not run tracked_users_reports migrations: %w", err)
+	}
+
+	return &UniqueUsersCollector{
+		log: log,
+		hllSettings: &hll.Settings{
+			Log2m:             config.GetInt("TrackedUsers.precision", 14),
+			Regwidth:          config.GetInt("TrackedUsers.registerWidth", 5),
+			ExplicitThreshold: hll.AutoExplicitThreshold,
+			SparseEnabled:     true,
+		},
+		instanceID: config.GetString("INSTANCE_ID", "1"),
+	}, nil
```

Comment:
pass a config.Config, don't use the global config package functions.

This will make it easier/safer to configure during tests. t.SetEnv is not safe on parallel executions.